lib/src/kbasesearchengine/main/NarrativeInfoDecorator.java

```diff
@@ -96,51 +107,69 @@ public SearchObjectsOutput searchObjects(final SearchObjectsInput params, final
             final List<ObjectData> objects,
-            final Map<Long, Tuple5 <String, Long, Long, String, String>> accessGroupNarrInfo) {
+            final Map<Long, Tuple5 <String, Long, Long, String, String>> accessGroupNarrInfo,
+            final List<String> removedGuids) {
 
         final Map<Long, Tuple5 <String, Long, Long, String, String>> retVal = new HashMap<>();
 
         if (accessGroupNarrInfo != null) {
             retVal.putAll(accessGroupNarrInfo);
         }
-        final Set<Long> wsIdsSet = new HashSet<>();
-
-        for (final ObjectData objData: objects) {
+        final Set<String> userNames = new HashSet<>();
+        final Iterator<ObjectData> iter = objects.iterator();
+        while (iter.hasNext()) {
+            final ObjectData objData = iter.next();
             final GUID guid = new GUID(objData.getGuid());
             if (WorkspaceEventHandler.STORAGE_CODE.equals(guid.getStorageCode())) {
-                wsIdsSet.add((long) guid.getAccessGroupId());
-            }
-        }
-        final Set<String> userNames = new HashSet<>();
-        for (final long workspaceId: wsIdsSet) {
-            final NarrativeInfo narrInfo = narrInfoProvider.findNarrativeInfo(workspaceId);
-            if (narrInfo != null) {
-                final Tuple5<String, Long, Long, String, String> tempNarrInfo =
-                        new Tuple5<String, Long, Long, String, String>()
-                                .withE1(narrInfo.getNarrativeName())
-                                .withE2(narrInfo.getNarrativeId())
-                                .withE3(narrInfo.getTimeLastSaved())
-                                .withE4(narrInfo.getWsOwnerUsername());
-                userNames.add(tempNarrInfo.getE4());
-                retVal.put(workspaceId, tempNarrInfo);
-            }
-            else {
-                retVal.put(workspaceId, null);
+                final long workspaceId = guid.getAccessGroupId();
+                final NarrativeInfo narrInfo = narrInfoProvider.findNarrativeInfo(workspaceId);
+
+                if (narrInfo != null) {
+                    final Tuple5<String, Long, Long, String, String> tempNarrInfo =
+                            new Tuple5<String, Long, Long, String, String>()
+                                    .withE1(narrInfo.getNarrativeName())
+                                    .withE2(narrInfo.getNarrativeId())
+                                    .withE3(narrInfo.getTimeLastSaved())
+                                    .withE4(narrInfo.getWsOwnerUsername());
+                    userNames.add(tempNarrInfo.getE4());
+                    retVal.put(workspaceId, tempNarrInfo);
+                }
+                else {
+                    iter.remove();
+                    removedGuids.add(objData.getGuid());
+
+                }
+            //if workspace is not a narrative, remove results from search
+            } else{
+                iter.remove();
```

Comment:
This appears to remove any object where the storage code is not WS, which is not behavior we want - I don't see how it has anything to do with whether a workspace is a narrative or not.

Also, it seems that if we don't want to show non-narrative workspaces in search results (which I think we do, for refdata if nothing else) we should just not index them.

Reply:
I think I missed the conversation about ref data. I can remove this.